src/cft/cft.jl

```diff
@@ -0,0 +1,162 @@
+        FwT0 = (k,κ) -> [fmultiplier(m) ≈ 0 ? (tFwT0(m-eps(),mm-eps())+tFwT0(m+eps(),mm+eps()))/2 : tFwT0(m,mm)  for (m,mm) in zip(k,κ)]
+    else
+        FwT0 = (k,κ) -> tFwT0(k,κ)
+    end
+    
+    # If stabilise is false, then we compute the supporter functions associated with ̃U₀ and V₁ (U0 and wT1)
+    # otherwise if it's true, then we compute the support functions asscoiated with ̃Uₙ₊₁ and Vₙ₊₂. 
+
+    ## Define function F[wT1] / (λ - i*μ*sgn(k)+ iηk + abs.(k))
+    if stabilise == false
+        tFwT1 = (k, κ) -> -im * pi * besselj.(1, k) ./ fmultiplier(κ)
+    else
+        tFwT1 = (k, κ) -> (-im).^(N+2) * pi * besselj.(N+2, k) ./ fmultiplier(κ)
+    end
+    
+    if λ ≈ 0
+        FwT1 = (k, κ) -> [fmultiplier(m) ≈ 0 ? (tFwT1(m-eps(),mm-eps())+tFwT1(m+eps(),mm+eps()))/2 : tFwT1(m,mm)  for (m,mm) in zip(k,κ)]
+    else
+        FwT1 = (k,κ) -> tFwT1(k,κ)
+    end
+    
+    ## Define function F[U0] / (λ - i*μ*sgn(k)+ iηk + abs.(k))
+    if stabilise == false
+        tFU0 = (k,κ) -> (pi * besselj.(1, abs.(k)) + 2 .*sin.(k)./k - 2 .* sin.(abs.(k)) ./ abs.(k)) ./ fmultiplier(κ)
+    else
+        tFU0 = (k,κ) -> ( (-im).^(N+2)*pi.*besselj.(N+2, k) ) ./ hfmultiplier(κ)
+    end
+    FU0 = (k,κ) -> [m ≈ 0 ? (tFU0(m-eps(),mm-eps())+tFU0(m+eps(),mm+eps()))/2 : tFU0(m,mm)  for (m,mm) in zip(k,κ)]
+    
+    ## Define function F[U-1] / (λ - i*μ*sgn(k)+ iηk + abs.(k))
+    tFU_1 = (k,κ) -> ( im*pi*k.*besselj.(0,abs.(k)) ./ abs.(k)) ./ fmultiplier(κ)
+    FU_1 = (k,κ) -> [m ≈ 0 ? (tFU_1(m-eps(),mm-eps())+tFU_1(m+eps(),mm+eps()))/2 : tFU_1(m,mm)  for (m,mm) in zip(k,κ)]
+    
+    ## Different element supporter functions on the same sized elements are simply translations
+    ## of the supporter functions on a reference element s*[-1,1] where s is the scaling.
+    ## Hence, we only want to do 4 FFTs per differently sized element to compute
+    ## the supporter functions on the reference elements s*[-1,1]. These are then translated in
+    ## interpolate_supporter_functions. 
+    
+    # Independent variable in physical space
+    x = ifftshift((fftfreq(length(ω), 1/δ)) * 2 * pi) 
+    
+    # Compute reference supporter functions
+    ywT0 = []; ywT1 = []; yU0 = []; yU_1 = []
+    for ss in s
+        sFwT0 = k -> (1/ss).*FwT0((1/ss) .*k, k)
+        sFwT1 = k -> (1/ss).*FwT1((1/ss) .*k, k)
+        sFU0 = k -> (1/ss).*FU0((1/ss) .*k, k)
+        sFU_1 = k -> (1/ss).*FU_1((1/ss) .*k, k)
+        append!(ywT0, [cifft(sFwT0, ω, δ, W, x)])
+        append!(ywT1, [cifft(sFwT1, ω, δ, W, x)])
+        append!(yU0, [cifft(sFU0, ω, δ, W, x)])
+        append!(yU_1, [cifft(sFU_1, ω, δ, W, x)])
+    end
+
+    return (x, (ywT0, yU_1, ywT1, yU0)) 
+end
+
+# FFT approximation of the inverse Fourier Transforms
+# This approximates (1/2π) ∫ f(ω)exp(i ω x) dω. 
+function cifft(f::Function, ω::AbstractVector, δ::Real, W::Real, x::AbstractVector)
+    yf= ifftshift(ifft(f(ω)))
+    N = length(ω)
+    yf = (δ .* N .* exp.(-im .*x .*W)  ./ (2*pi)) .* yf

```

Comment:
Don't assign to a variable and immediately return:
```suggestion
return (δ .* N .* exp.(-im .*x .*W)  ./ (2*pi)) .* yf
```